src/Modules/Visualization/ShowUncertaintyGlyphs.cc

```diff
@@ -0,0 +1,320 @@
+using namespace Algorithms;
+using namespace Geometry;
+using namespace Graphics;
+using namespace Graphics::Datatypes;
+
+MODULE_INFO_DEF(ShowUncertaintyGlyphs, Visualization, SCIRun);
+
+namespace SCIRun {
+namespace Modules {
+namespace Visualization {
+Dyadic3DTensor scirunTensorToEigenTensor(const Geometry::Tensor& t)
+{
+  Dyadic3DTensor newTensor(t.xx(), t.xy(), t.xz(), t.yy(), t.yz(), t.zz());
+  return newTensor;
+}
+
+Tensor eigenTensorToScirunTensor(const Dyadic3DTensor& t)
+{
+  return Tensor(t(0, 0), t(1, 0), t(2, 0), t(1, 1), t(2, 1), t(2, 2));
+}
+
+class ShowUncertaintyGlyphsImpl
+{
+ public:
+  ShowUncertaintyGlyphsImpl();
+  GeometryHandle run(const GeometryIDGenerator& idgen, const FieldHandle mean, const MatrixHandle covariance, std::string& idname);
+  ShowUncertaintyGlyphsImpl& setShowTensors(bool showTensors);
+  ShowUncertaintyGlyphsImpl& setTensorsResolution(int resolution);
+  ShowUncertaintyGlyphsImpl& setTensorsTransparency(bool useTransparency);
+  ShowUncertaintyGlyphsImpl& setTensorsUniformTransparencyValue(double transparency);
+  ShowUncertaintyGlyphsImpl& setTensorsSuperquadricEmphasis(double emphasis);
+  ShowUncertaintyGlyphsImpl& setTensorsScale(double scale);
+
+ private:
+  void verifyData(const FieldList& fields);
+  void getPoints(const FieldHandle& fields);
+  void getPointsForField(FieldHandle field, std::vector<int>& indices, std::vector<Point>& points);
+  void computeOffsetSurface(FieldHandle mean, MatrixHandle covariance);
+
+  GlyphConstructor constructor_;
+  int fieldCount_ = 0;
+  int fieldSize_ = 0;
+
+  std::vector<Point> points_;
+  std::vector<int> indices_;
+  std::vector<Dyadic3DTensor> meanTensors_;
+  std::vector<Eigen::Matrix<double, 6, 6>> covarianceMatrices_;
+
+  bool showTensors_ = true;
+  double emphasis_ = 3.0;
+  double scale_ = 1.0;
+  bool useTensorTransparency_ = true;
+  double transparency_ = 0.5;
+  int resolution_ = 101;
+};
+
+ShowUncertaintyGlyphsImpl::ShowUncertaintyGlyphsImpl()
+{
+  constructor_ = GlyphConstructor();
+}
+
+ShowUncertaintyGlyphsImpl& ShowUncertaintyGlyphsImpl::setShowTensors(bool showTensors)
+{
+  showTensors_ = showTensors;
+  return *this;
+}
+
+ShowUncertaintyGlyphsImpl& ShowUncertaintyGlyphsImpl::setTensorsResolution(int resolution)
+{
+  resolution_ = resolution;
+  return *this;
+}
+
+ShowUncertaintyGlyphsImpl& ShowUncertaintyGlyphsImpl::setTensorsTransparency(bool useTransparency)
+{
+  useTensorTransparency_ = useTransparency;
+  return *this;
+}
+
+ShowUncertaintyGlyphsImpl& ShowUncertaintyGlyphsImpl::setTensorsUniformTransparencyValue(double transparency)
+{
+  transparency_ = transparency;
+  return *this;
+}
+
+ShowUncertaintyGlyphsImpl& ShowUncertaintyGlyphsImpl::setTensorsSuperquadricEmphasis(double emphasis)
+{
+  emphasis_ = emphasis;
+  return *this;
+}
+
+ShowUncertaintyGlyphsImpl& ShowUncertaintyGlyphsImpl::setTensorsScale(double scale)
+{
+  scale_ = scale;
+  return *this;
+}
+
+
+// TODO move
+enum FieldDataType
```

Comment:
There's already an enum for this, can you use that one?

Reply:
It's in 2 other modules. I was planning on moving this, just don't know where to put it

Reply:
You can reuse `Mesh::NODES_E, Mesh::FACES_E`, and the like instead.